src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.BulldozerProducer;
+import core.mate.academy.producer.ExcavatorProducer;
+import core.mate.academy.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> list = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            list = new ArrayList<>(new BulldozerProducer().get());
+        }
```

Comment:
You should remove the unnecessary if-else structure after the return statement. Since you're returning a list inside the if blocks, there is no need for an else block. Simplify the structure to make the code cleaner and to comply with the checklist.